### PR TITLE
Setup connection to sqlite

### PR DESCRIPTION
**Pull request by jomei**
#6

### CODE REVIEW

**Comment by jomei** [comment]
@Yapanyushin, I waiting your task with Bill CRUD. Let's merge this taske after finishing your one.

File: app/src/main/java/billbalancer/app/BillFragment.java [hunk 1]
```diff
@@ -22,6 +23,13 @@ public void onCreate(Bundle savedInstanceState) {
         mBill = new Bill();
     }
 
+    @Override
+    public void onPause(){
+        super.onPause();
+        BillRepository.getInstance(getActivity())
+                .update(mBill);
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/BillFragment.java]: Why we need update Bill in repository onPause? This method calls even if device blocked, as i know. Is it the behavior what we excpect?

File: app/src/main/java/billbalancer/app/db/DatabaseHelper.java [hunk 2]
```diff
@@ -4,31 +4,28 @@
...
 
     @Override
-    public void onCreate(SQLiteDatabase db) {
-        db.execSQL("create table " + BILLS_TABLE_NAME +" ("
-                + KEY_ID + " integer BILLS_TABLE_NAME key autoincrement,"
-                + KEY_NAME + " text);");
+    public void onCreate(SQLiteDatabase db){
+        db.execSQL("CREATE TABLE " + BillTable.NAME + "(" +
+            " _id integer primary key autoincrement, " +
+            BillTable.Cols.UUID + ", " +
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/db/DatabaseHelper.java]: Why we need UUID and _id? Looks like we have data duplication in table.

File: app/src/main/java/billbalancer/app/model/Bill.java [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+public class Bill extends Model {
+    private String mTitle;
     private Double mTips;
     private Double mTotal;
 
+    public Bill(UUID id) {
+        super(id);
+    }
+
+    public Bill(){
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/model/Bill.java]: Is this method necessary?

File: app/src/main/java/billbalancer/app/model/Bill.java [hunk 4]
```diff
@@ -22,4 +42,6 @@ public void setTotal(Double total){
     public Double getTotal() {
         return mTotal;
     }
+
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/model/Bill.java]: Remove empty rows

File: app/src/main/java/billbalancer/app/BillActivity.java [hunk 5]
```diff
@@ -29,26 +27,28 @@ public void onPartSelected(int position) {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_bill);
-        boolean isBillCreate = getIntent().getBooleanExtra(BILL_CREATE, false);
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/BillActivity.java]: This is looks wrong. What if we want to start this for any other reason then create or update?

Comment (jomei) [inline app/src/main/java/billbalancer/app/BillActivity.java]: Now we dont have any other reason

File: app/src/main/java/billbalancer/app/model/Bill.java [hunk 6]
```diff
@@ -21,8 +20,10 @@ public Bill(UUID id) {
 
     @Override
     public boolean equals(Object other) {
-        Bill otherBill = (Bill) other;
-        return this.mId == otherBill.getId();
+        return other instanceof Bill &&
+                this.mId.toString().equals(
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/model/Bill.java]: Do we need to compare string representations of ID field?  As i understand we have integer data types, am i right?

Comment (jomei) [inline app/src/main/java/billbalancer/app/model/Bill.java]: Currently our ids are UUID.toString()

File: app/src/main/java/billbalancer/app/BillActivity.java (hunk 5 above)
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/BillActivity.java]: It means if we will have in future, we have to keep in mind that new Bill will always be created if we start this Activity.

Comment (jomei) [inline app/src/main/java/billbalancer/app/BillActivity.java]: If we'll have something in the future you anyway should come here and change this code.
Our sources must be as simple as possible and every unrequired piece of code must be thrown out

File: app/src/main/java/billbalancer/app/BillListFragment.java [hunk 7]
```diff
@@ -46,13 +45,23 @@ public void onClick(View view) {
...
+    public void onResume() {
+        super.onResume();
+        updateUI();
+    }
 
-    private void updateUI(){
-        BillStorage storage = BillStorage.getInstance(getActivity());
-        List<Bill> bills = storage.getBills();
+    private void updateUI() {
+        BillRepository repository = (BillRepository) BillRepository.getInstance(getActivity());
```
Comment (jomei) [inline app/src/main/java/billbalancer/app/BillListFragment.java]: Unrequired `(BillRepository) `

File: app/src/main/java/billbalancer/app/model/Repository/AbstractRepository.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package billbalancer.app.model.repository;
```
Comment (Yapanyushin) [inline app/src/main/java/billbalancer/app/model/Repository/AbstractRepository.java]: All packages started with underline, Repository not. It works, but on linux it looks bad. Also this makes disorder

**Pull request merged**